docs/DeveloperGuide.adoc

```diff
@@ -337,46 +397,232 @@ Use case ends.
-Windows, Linux, Unix, OS-X
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. MoneyGoWhere shows an error message.
++
+Use case resumes at step 2.
 
-[[private-contact-detail]] Private contact detail::
-A contact detail that is not meant to be shared with others
+'''
+[discrete]
+=== UC04 Find a spending
 
-[appendix]
-== Product Survey
+*MSS*
 
-*Product Name*
+1.  User requests to find all occurrences of a keyword or within a cost range or date range in a spending.
+2.  MoneyGoWhere returns any spending found with the keyword or within a cost range or date range specified.
++
+Use case ends.
 
-Author: ...
+'''
+[discrete]
+=== UC05 Set a monthly budget
+
+*MSS*
 
-Pros:
+1.  User requests to set a monthly budget
+2.  MoneyGoWhere sets a specific monthly budget
++
+Use case ends.
 
-* ...
-* ...
+*Extensions*
 
-Cons:
+[none]
+* 1a. The given budget value is invalid.
++
+[none]
+** 1a1. MoneyGoWhere shows an error message.
++
+Use case resumes at step 1.
+[none]
 
-* ...
-* ...
+'''
+[discrete]
+=== UC06 Generate statistics
+
+*MSS*
+
+1.  User requests to generate statistics.
+2.  MoneyGoWhere generates statistical analysis such as total spending of each tag.
++
+Use case ends.
+[none]
+
+'''
+[discrete]
+=== UC07 List all tags
+
+*MSS*
+
+1.  User requests to list all the recorded tags.
+2.  MoneyGoWhere lists all the recorded tags.
++
+Use case ends.
+[none]
+
+'''
+[discrete]
+=== UC08 Import external data from a CSV file
+
+*MSS*
+
+1.  User requests to import data from CSV file by specifying the file path.
+2.  MoneyGoWhere imports the data from the CSV file located at specified file path.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The file path is invalid.
++
+[none]
+** 1a1. MoneyGoWhere shows an error message.
++
+Use case resumes at step 1.
+[none]
+
+'''
+[discrete]
+=== UC09 Export data to a CSV file
+
+*MSS*
+
+1.  User requests to export data from CSV file by specifying the file path.
+2.  MoneyGoWhere exports the data into the CSV file located at specified file path.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The file path is invalid.
++
+[none]
+** 1a1. MoneyGoWhere shows an error message.
++
+Use case resumes at step 1.
+[none]
+
+'''
+[discrete]
+=== UC10 Undo a command
+
+*MSS*
+
+1.  User requests to undo a command.
+2.  MoneyGoWhere restores the list of spending to the state before the previous command was executed.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. There are no undo commands executed previously.
++
+[none]
+** 1a1. MoneyGoWhere shows an error message.
++
+Use case ends.
+[none]
+
+'''
+[discrete]
+=== UC11 Redo a command
+
+*MSS*
+
+1.  User requests to redo a command.
+2.  MoneyGoWhere executes the previous command that was undone.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. There are no undo commands executed previously.
++
+[none]
+** 1a1. MoneyGoWhere shows an error message.
++
+Use case ends.
+[none]
+
+'''
+[discrete]
+=== UC12 Graphing the data
```

Comment:
You may consider specifying the range of data used for the graph e.g. a month/year/etc.